src/kvstore/comm.h

```diff
@@ -670,31 +715,75 @@ class CommDevice : public Comm {
   using KeyAttrs = std::tuple<int, TShape, int>;
   // try to allocate buff on device evenly
   void InitMergeBuffer(const std::vector<Context>& devs) {
-    std::sort(sorted_key_attrs_.begin(), sorted_key_attrs_.end(), [](
-              const KeyAttrs& a, const KeyAttrs& b) {
-      return std::get<1>(a).Size() > std::get<1>(b).Size();
-    });
-
-    std::unordered_map<int, std::pair<Context, size_t>> ctx_info;
-    for (auto d : devs) {
-      ctx_info[d.dev_id] = std::make_pair(d, 0);
+    std::sort(sorted_key_attrs_.begin(), sorted_key_attrs_.end(),
+              [](const KeyAttrs& a, const KeyAttrs& b) {
+                return std::get<1>(a).Size() > std::get<1>(b).Size();
+              });
+
+    std::vector<Context> g1, g2;
```

Comment:
Are there more readable var names for `g1` and `g2` to explain its purpose?

Reply:
So g represents group here. It means I separate the GPU cards into two communication groups.

Reply:
I've added some description for g1 and g2, and moved them to class members.